linux_os/guide/system/software/integrity/fips/enable_fips_mode/rule.yml

```diff
@@ -0,0 +1,46 @@
+documentation_complete: true
+
+prodtype: rhel8,fedora
+
+title: Enable FIPS Mode
+
+description: |-
+    To enable FIPS mode, run the following command:
+    <pre>fips-mode-setup --enable</pre>
+    <br />
+    On a system running in FIPS mode, the kernel FIPS mode flag
+    (<tt>/proc/sys/crypto/fips_enabled</tt>) should be set to <tt>1</tt> and
+    the <tt>/etc/system-fips</tt> should exist. The system crypto policy should
+    be set to FIPS in <tt>/etc/crypto-policies/config</tt>. Also, the Dracut
+    <tt>fips</tt> module should be loaded. Furthermore, the system running in
+    FIPS mode should be FIPS certified.
```

Comment:
This paragraph should be removed because it reads as if more configurations need to take place rather than just running `fips-mode-setup --enable`

Reply:
it would be nice to have described what the command will do, like what settings are necessary to be in FIPS mode.

Reply:
Most people don't care about what the command will do. Just that it does as advertised. We don't describe all the steps of running an rpm installation or update or the grub2-mkconfig command. 

Still, the way it is written isn't in the form of describing what the command will do but rather additional tasks.

Reply:
> We don't describe all the steps of running an rpm installation or update or the grub2-mkconfig command.

Well, there is no need for a pseudo-algorithm of the command.

@redhatrises My rational for advocating for description of the expected FIPS state, is that it can help when the rule fails. Sometimes it can be hard to know what may be going wrong.

In this case, the remediation is simple as running a command, there is not much room for mistakes here. But the OVAL definition checks a lot of stuff, and with these described it is easier to check and debug, or for non-experts in OVAL to understand what the rule expects.

There are cases where check and remediation are complex, like in audit rules, for example. When rule fails scan, with description in rule, people can go and look up into their audit rules and compare what is expected to be the there, and what is actually there.

Reply:
> > We don't describe all the steps of running an rpm installation or update or the grub2-mkconfig command.
> 
> Well, there is no need for a pseudo-algorithm of the command.
> 
> @redhatrises My rational for advocating for description of the expected FIPS state, is that it can help when the rule fails. Sometimes it can be hard to know what may be going wrong.

That's what `--oval-results` is for.

> In this case, the remediation is simple as running a command, there is not much room for mistakes here. But the OVAL definition checks a lot of stuff, and with these described it is easier to check and debug, or for non-experts in OVAL to understand what the rule expects.
> 
> There are cases where check and remediation are complex, like in audit rules, for example. When rule fails scan, with description in rule, people can go and look up into their audit rules and compare what is expected to be the there, and what is actually there.

Anyway the wording needs to be rewritten in such a way so that it doesn't read as there are addition steps to follow.
For example:

```
    The <tt>fips-mode-setup</tt> command will configure the system in 
    FIPS mode by automatically configuring the following:
    <ul>
    <li>Setting <tt>/proc/sys/crypto/fips_enabled</tt> to <tt>1</tt></li>
    <li>Creating <tt>/etc/system-fips</tt></li>
    <li>Enabling FIPS in the system crypto policy in <tt>/etc/crypto-policies/config</tt></li>
    <li>Loading the Dracut <tt>fips</tt> module</li> 
    </ul>
    Furthermore, the system running in FIPS mode should be FIPS
    certified by NIST.
```

Reply:
`Furthermore, the system running in FIPS mode should be FIPS certified by NIST.`
This is a misleading sentence. I'd suggest removing it or replacing with something like:
`Furthermore for the system to be FIPS compliant the components implementing cryptography have to be FIPS certified by NIST.`